pandas/tests/indexes/test_multi.py

```diff
@@ -3279,3 +3279,78 @@ def test_duplicate_multiindex_labels(self):
         with pytest.raises(ValueError):
             ind.set_levels([['A', 'B', 'A', 'A', 'B'], [2, 1, 3, -2, 5]],
                            inplace=True)
+
+    def test_repr(self):
+        # GH21145
+
```

Comment:
set the context manager for width here (so need multiple tests for this), e.g. with a small value then a large value to see how it works on different settings.